File: engine/engine-model/src/main/scala/server/movement/RelativeLifeMovement.scala
package server.movement

import java.awt.Point

import tools.data.output.LittleEndianWriter

class RelativeLifeMovement(private var _type: Int, private var _position: Point, private var _duration: Int, private var _newState: Int) extends AbstractLifeMovement(_type = _type, _position = _position, _duration = _duration, _newState = _newState) {
  override def serialize(lew: LittleEndianWriter): Unit = {
    lew.write(theType)
    lew.writeShort(position.x)
    lew.writeShort(position.y)
    lew.write(newState)
    lew.writeShort(duration)
  }
}

File: engine/engine-model/src/main/java/server/movement/RelativeLifeMovement.java
package server.movement;

import java.awt.Point;

import tools.data.output.LittleEndianWriter;

public class RelativeLifeMovement extends AbstractLifeMovement {
   public RelativeLifeMovement(int theType, Point position, int duration, int newState) {
      super(theType, position, duration, newState);
   }

   @Override
   public void serialize(LittleEndianWriter lew) {
      lew.write(theType());
      lew.writeShort(position().x);
      lew.writeShort(position().y);
      lew.write(newState());
      lew.writeShort(duration());
   }
}
